extensions/auth/portability-auth-smugmug/src/main/java/org/dataportabilityproject/auth/smugmug/SmugMugAuthDataGenerator.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018 The Data Transfer Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dataportabilityproject.auth.smugmug;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import org.dataportabilityproject.auth.smugmug.SmugMugAuth.SmugMugOauthInterface;
+import org.dataportabilityproject.spi.api.auth.AuthDataGenerator;
+import org.dataportabilityproject.spi.api.auth.AuthServiceProviderRegistry.AuthMode;
+import org.dataportabilityproject.spi.api.types.AuthFlowConfiguration;
+import org.dataportabilityproject.types.transfer.auth.AppCredentials;
+import org.dataportabilityproject.types.transfer.auth.AuthData;
+import org.dataportabilityproject.types.transfer.auth.TokenSecretAuthData;
+import org.scribe.model.Verifier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/* SmugmugAuthDataGenerator used for obtaining auth credentials for the Smugmug API*/
+public class SmugMugAuthDataGenerator implements AuthDataGenerator {
+  private final Logger logger = LoggerFactory.getLogger(SmugMugAuthDataGenerator.class);
+  private final String perms;
+  private final SmugMugOauthInterface smugMugOauthInterface;
+
+  public SmugMugAuthDataGenerator(AppCredentials appCredentials, AuthMode authMode) {
+    this.perms = authMode == AuthMode.IMPORT ? "Add" : "Read";
+    this.smugMugOauthInterface =
+        new SmugMugOauthInterface(appCredentials.getKey(), appCredentials.getSecret());
+  }
+
+  @Override
+  public AuthFlowConfiguration generateConfiguration(String callbackBaseUrl, String id) {
+    // Generate a request token and include that as initial auth data
+    TokenSecretAuthData authData = null;
+    try {
+      authData = smugMugOauthInterface.getRequestToken(callbackBaseUrl + "/callback1/smugmug");
+    } catch (IOException e) {
+      logger.debug("Couldnt get authData {}", e.getMessage());
+      return null;
```

Comment:
Should we just rethrow

Reply:
Hm I wanted to be consistent with FlickrAuthDataGenerator and RTM AuthDataGenerator, but it looks like those two aren't consistent with instagram, microsoft or google auth data generators. 

I can create a separate PR To throw a new Runtime or IllegalArg exception but this probably also has to do with the discussion in #176 .